include/rmm/device_uvector.hpp

```diff
@@ -0,0 +1,262 @@
+   *
+   * @param new_size The desired number of elements
+   * @param stream The stream on which to perform the allocation/copy (if any)
+   */
+  void resize(std::size_t new_size, cudaStream_t stream = 0)
+  {
+    _storage.resize(elements_to_bytes(new_size), stream);
+  }
+
+  /**
+   * @brief Forces deallocation of unused device memory.
+   *
+   * If `capacity() > size()`, reallocates and copies vector contents to eliminate unused memory.
+   *
+   * @param stream Stream on which to perform allocation and copy
+   */
+  void shrink_to_fit(cudaStream_t stream = 0) { _storage.shrink_to_fit(stream); }
+
+  /**
+   * @brief Release ownership of device memory storage.
+   *
+   * @return The `device_buffer` used to store the vector elements
+   */
+  device_buffer release() noexcept { return std::move(_storage); }
+
+  /**
+   * @brief Returns the number of elements that can be held in currently allocated storage.
+   *
+   * @return std::size_t The number of elements that can be stored without requiring a new
+   * allocation.
+   */
+  std::size_t capacity() const noexcept { return bytes_to_elements(_storage.capacity()); }
+
+  /**
+   * @brief Returns pointer to underlying device storage.
+   *
+   * @return Raw pointer to element storage in device memory.
+   */
+  pointer data() noexcept { return static_cast<pointer>(_storage.data()); }
+
+  /**
+   * @brief Returns const pointer to underlying device storage.
+   *
+   * @return const_pointer Raw const pointer to element storage in device memory.
+   */
+  const_pointer data() const noexcept { return static_cast<const_pointer>(_storage.data()); }
+
+  /**
+   * @brief Returns an iterator to the first element.
+   *
+   * @return Iterator to the first element.
+   */
+  iterator begin() noexcept { return data(); }
+
+  /**
+   * @brief Returns an iterator to the first element.
+   *
+   * @return Iterator to the first element.
+   */
+  const_iterator begin() const noexcept { return data(); }
+
+  /**
+   * @brief Returns an iterator to the element following the last element of the vector.
+   *
+   * The element referenced by `end()` is a placeholder and dereferencing it results in undefined
+   * behavior.
+   *
+   * @return Iterator to one past the last element.
+   */
+  iterator end() noexcept { return data() + size(); }
+
+  /**
+   * @brief Returns an iterator to the element following the last element of the vector.
+   *
+   * The element referenced by `end()` is a placeholder and dereferencing it results in undefined
+   * behavior.
+   *
+   * @return Iterator to one past the last element.
+   */
+  const_iterator end() const noexcept { return data() + size(); }
+
+  /**
+   * @brief Returns the number of elements in the vector.
+   *
+   * @return The number of elements.
+   */
+  std::size_t size() const noexcept { return bytes_to_elements(_storage.size()); }
+
+  /**
+   * @brief Returns if the vector contains no elements, i.e., `size() == 0`.
```

Comment:
It returns in either case. :)

```suggestion
   * @brief Returns true if the vector contains no elements, i.e., `size() == 0`.
```